sdk/cosmos/azure-cosmos-encryption/src/main/java/com/azure/cosmos/encryption/models/SqlQuerySpecWithEncryption.java

```diff
@@ -75,7 +75,9 @@ public void addEncryptionParameter(String path, SqlParameter sqlParameter) {
                                 System.arraycopy(cipherText, 0, cipherTextWithTypeMarker, 1, cipherText.length);
                                 SqlParameter encryptedParameter = new SqlParameter(sqlParameter.getName(),
                                     cipherTextWithTypeMarker);
-                                this.sqlQuerySpec.getParameters().add(encryptedParameter);
+                                if (!parameters.contains(encryptedParameter)) {
```

Comment:
We are encrypting sql parameter via SqlQuerySpecMono  https://github.com/Azure/azure-sdk-for-java/blob/525606e31c75ec45fb3e8cd1d0e42272756b71c0/sdk/cosmos/azure-cosmos-encryption/src/main/java/com/azure/cosmos/encryption/CosmosEncryptionAsyncContainer.java#L291 
So on retry it will add duplicates

Reply:
we should try to rely on the existing query parameters. why should we attempt to re-build the query parameters again?

either we should throw the query parameters altogether and rebuild or re-use the existing ones.

why can't we do that?

Reply:
done, changed the code around to reuse query spec . Didn't do it first place due to the code readability for retry.